src/main/org/codehaus/groovy/transform/stc/StaticTypeCheckingSupport.java

```diff
@@ -1575,16 +1575,18 @@ private static boolean equalIncludingGenerics(GenericsType orig, GenericsType co
         if (!equalIncludingGenerics(orig.getType(), copy.getType())) return false;
         ClassNode lower1 = orig.getLowerBound();
         ClassNode lower2 = copy.getLowerBound();
-        if ((lower1==null || lower2 ==null) && lower1!=lower2) return false;
-        if (lower1==lower2) return true;
```

Comment:
This is the shortcut that's incorrect if `lower1` and `lower2` are both null. The other changes are there mainly for consistency.